modules/vault-cluster/main.tf

```diff
@@ -178,3 +178,33 @@ data "aws_iam_policy_document" "instance_role" {
     }
   }
 }
+
+resource "aws_s3_bucket" "vault_storage" {
+  count         = "${var.enable_s3_backend ? 1 : 0}"
+  bucket        = "${var.s3_bucket_name}"
+  force_destroy = "${var.force_destroy_s3_bucket}"
+
+  tags {
+    Description = "Used for secret storage with Vault. DO NOT DELETE this Bucket unless you know what you are doing."
+  }
+}
+
+resource "aws_iam_role_policy" "vault_s3" {
+  count  = "${var.enable_s3_backend ? 1 : 0}"
+  name   = "vault_s3"
+  role   = "${aws_iam_role.instance_role.id}"
+  policy = "${data.aws_iam_policy_document.vault_s3.json}"
```

Comment:
Since this `aws_iam_policy_document` uses `count`, the recommended syntax in Terraform 0.11 is: `element(concat(data.aws_iam_policy_document.vault_s3.*.json, list("")), 0)`.